src/ceph-volume/ceph_volume/util/prepare.py

```diff
@@ -458,9 +459,23 @@ def osd_mkfs_bluestore(osd_id, fsid, keyring=None, wal=False, db=False):
 
     command = base_command + supplementary_command
 
-    _, _, returncode = process.call(command, stdin=keyring, show_command=True)
-    if returncode != 0:
-        raise RuntimeError('Command failed with exit code %s: %s' % (returncode, ' '.join(command)))
+    """
+    When running in containers the --mkfs on raw device sometimes fails
+    to acquire a lock through flock() on the device because systemd-udevd holds one temporarily.
+    See KernelDevice.cc and _lock() to understand how ceph-osd acquires the lock.
+    Because this is really transient, we retry up to 5 times and wait for 1 sec in-between
+    """
+    for retry in range(5):
```

Comment:
Maybe we should expose an environment variable so we can tweak the retry value here instead of hardcoding 5. We do something similar in the zap command: https://github.com/ceph/ceph/blob/master/src/ceph-volume/ceph_volume/devices/lvm/zap.py#L28

Thoughts?

Reply:
Do you have a use case in mind for this? Testing or something?

I'm all for flexibility but I dislike adding features no one uses even more ;)

Reply:
Mainly for flexibility, nothing to block the PR for. My thought was if we came across a system that needed a longer delay then we wouldn't need to change code to allow for that.